validator/sawtooth_validator/gossip/structure_verifier.py

```diff
@@ -72,10 +73,33 @@ def is_valid_batch(batch):
                          "order of transactions in the batch: %s txn: %s",
                          batch.header_signature, header_txn_id)
             return False
+        txn_header = _deserialize_txn_header(txn.header)
+        for dep in txn_header.dependencies:
+            if not dep:
+                LOGGER.debug("Malformed dependency in batch: %s, txn:%s",
```

Comment:
This method could probably be more descriptive; this really just reports when a dependency is empty, not any other malformed case.